pkg/controller/eventing/inmemory/channel/reconcile.go

```diff
@@ -140,13 +154,15 @@ func (r *reconciler) reconcile(ctx context.Context, c *eventingv1alpha1.Channel)
 	svc, err := util.CreateK8sService(ctx, r.client, c)
 	if err != nil {
 		logger.Info("Error creating the Channel's K8s Service", zap.Error(err))
+		r.recorder.Eventf(c, corev1.EventTypeWarning, k8sServiceCreateFailed, "Failed to create Channel's K8s Service: %v", err)
 		return err
 	}
 	c.Status.SetAddress(controller.ServiceHostName(svc.Name, svc.Namespace))
 
 	_, err = util.CreateVirtualService(ctx, r.client, c, svc)
 	if err != nil {
 		logger.Info("Error creating the Virtual Service for the Channel", zap.Error(err))
+		r.recorder.Eventf(c, corev1.EventTypeWarning, virtualServiceCreateFailed, "Failed to create Virtual Service for the Channel: %v", err)
```

Comment:
his is more appropriately 'Failed to reconcile Channel's VirtualService: %v", because it could be caused by an issue when Listing VirtualServices.